src/main/java/ai/elimu/dao/jpa/WordDaoJpa.java

```diff
@@ -24,6 +25,26 @@ public Word readByText(String text) throws DataAccessException {
         }
     }
 
+    @Override
+    public Word readByTextAndType(String text, WordType wordType) throws DataAccessException {
+        String query = "SELECT w " +
+                "FROM Word w " +
+                "WHERE w.text = :text ";
+        if (wordType == null) {
+            query += "AND w.wordType IS NULL AND :wordType IS NULL";
```

Comment:
```suggestion
            query += "AND w.wordType IS NULL";
```

Reply:
.setParameter("wordType", wordType) requires this parameter in JPQL query.

Reply:
> .setParameter("wordType", wordType) requires this parameter in JPQL query.
> 

I see. Can we solve it like this?:
```java
if (wordType != null) {
    query.setParameter("wordType", wordType);
}
```